[GCP] Add dataproc data stream (#3789)
elastic/integrations: packages/gcp/_dev/build/docs/dataproc.md

# Dataproc

## Metrics

The `dataproc` dataset fetches metrics from [Dataproc](https://cloud.google.com/dataproc/) in Google Cloud Platform. It contains all metrics exported from the [GCP Dataproc Monitoring API](https://cloud.google.com/monitoring/api/metrics_gcp#gcp-dataproc).

> You can specify a single region to fetch metrics like `us-central1`. Be aware that GCP Storage does not use zones so `us-central1-a` will return nothing. If no region is specified, it will return metrics from all buckets.

Reviewer: Is this reference to GCP storage appropriate here?

> You can specify a single region to fetch metrics like `us-central1`. Be aware that GCP Dataproc does not use zones so `us-central1-a` will return nothing. If no region is specified, it will return metrics from all buckets.

Reviewer: After reading this line, I was wondering if we can simply remove the `zone` configuration from the data stream. what do you think?

```suggestion
You can specify a single region to fetch metrics like `us-central1`. Be aware that GCP Dataproc is a regional service. If no region is specified, it will return metrics from all buckets.
```
